### PR TITLE
Refactor Specialized EmptyState migration with Logic Alignment and Test Updates

### LINKED ISSUE #5123: Phase 1: EmptyState Migration - Specialized Screens
# Phase 1: EmptyState Migration - Specialized Screens

1. **Part of:** #5087 - EmptyState Component Implementation
2. **PR Number:** 5 of 7
3. **Estimated Effort:** 3-4 hours
4. **Dependencies:** Issue #5120 must be completed first
5. This is a detailed guide for the tasks to perform. There may be mistakes, so you'll need to use your discretion in tackling the various code changes and PRs. Use your initiative to have the best outcome for the long term maintainability of the code base.

## Overview

Refactor specialized screens (PluginStore, Notifications, OrganizationFunds) with context-aware empty state implementations.

## Files to Modify

- `src/screens/PluginStore/components/PluginList.tsx`
- `src/screens/Notification/Notification.tsx`
- `src/screens/OrganizationFunds/OrganizationFunds.tsx`
- Corresponding test files

### PluginList.tsx - Context-Aware Messaging

This screen requires conditional message logic based on search and filter state:

```typescript
const getEmptyMessage = () => {
  if (searchTerm) return 'noPluginsFound';
  if (filterOption === 'installed') return 'noInstalledPlugins';
  return 'noPluginsAvailable';
};

<EmptyState
  icon="extension"
  message={getEmptyMessage()}
  description={searchTerm ? 'tryDifferentSearch' : 'explorePluginStore'}
  dataTestId="plugins-empty-state"
/>
```

### PluginList.tsx
- [ ] Replace inline styles empty state with EmptyState
- [ ] Implement conditional message logic externally
- [ ] Add appropriate icon (`extension` or `plug`)
- [ ] Context-aware description based on filter/search state
- [ ] Tests cover all message scenarios

### Notification.tsx
- [ ] Replace inline empty state with EmptyState
- [ ] Add notifications icon (`notifications_none`)
- [ ] Tests updated

### OrganizationFunds.tsx
- [ ] Replace Stack-based empty state with EmptyState
- [ ] Add funds icon (`account_balance`)
- [ ] Consider adding "Create Fund" action button
- [ ] Tests updated

### Simple Empty State
```tsx
<EmptyState
  icon="notifications_none"
  message="noNotifications"
  dataTestId="notifications-empty-state"
/>
```

### With Action Button
```tsx
<EmptyState
  icon="account_balance"
  message="noFundsYet"
  description="createFirstFund"
  action={{
    label: "createFund",
    onClick: openCreateFundModal,
    variant: "primary"
  }}
  dataTestId="funds-empty-state"
/>
```

## Links

- **Comprehensive Plan:** [Comment #3671736597](https://github.com/PalisadoesFoundation/talawa-admin/issues/5111#issuecomment-3671736597)
- **Parent Issue:** #5087
- **Dependency:** Issue #1

### PR DESCRIPTION
**Pull request by rawadhossain**
**What kind of change does this PR introduce?**
Refactor specialized screens (`PluginStore, Notifications, OrganizationFunds`) with context-aware empty state implementations.

**Issue Number:**
Fixes #5123

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**
- **Refactored 3 screens** (PluginStore, Notification, OrganizationFunds) by replacing inline/Stack-based empty-state UIs with the shared `EmptyState` component, adding appropriate icons (`ExtensionOutlined`, `NotificationsNone`, `AccountBalance`) and `dataTestId` attributes.

- **Implemented context-aware conditional messaging** in PluginList: dynamically displays different messages/descriptions based on search term vs. filter state (e.g., "No plugins found" with search vs. "No installed plugins" when filtering).

- **Added action button to OrganizationFunds** empty state that opens the create fund modal.

**Does this PR introduce a breaking change?**
No

### CodeRabbit AI Review
- [x] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [x] I have implemented or provided justification for each non-critical suggestion
- [x] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [x] I have written tests for all new changes/features
- [x] I have verified that test coverage meets or exceeds 95%
- [x] I have run the test suite locally and all tests pass

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

## Summary by CodeRabbit

* **New Features**
  * Added accessibility labels for notification actions
  * Introduced enhanced empty states with icons and messaging across the app
  * Added support for displaying and sorting archived funds

* **Improvements**
  * Improved plugin store search guidance and discovery messaging
  * Better loading state indicators and empty state UI
  * Enhanced spacing and visual layout refinements
  * Extended localization support across multiple languages

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/screens/OrganizationFunds/OrganizationFundsMocks.ts [hunk 1]
```diff
@@ -206,3 +206,81 @@ export const MOCKS_ERROR = [
...
+                  createdAt: '2024-06-21T00:00:00Z',
+                },
+              },
+            ],
+          },
+        },
+      },
+    },
+  },
+];
```
[the remark this answers is not in the document]
Comment (rawadhossain) [reply src/screens/OrganizationFunds/OrganizationFundsMocks.ts:253]: MOCKS_WITH_ARCHIVED is intentionally scoped to validate archived vs active handling without expanding the base MOCKS fixture. This keeps the primary dataset minimal while still providing targeted coverage where isArchived behavior is exercised.

**Comment by rawadhossain** [comment]
@coderabbitai full review and reevaluate all resolved items and include the following points:

1. **Test Coverage & Quality Validation:**
   
   1. Validate whether any updated test files:
      
      1. Cover all use cases including edge cases, error scenarios, and success paths
      2. Have valid, properly structured tests following Jest and React Testing Library best practices
      3. Provide 100% test code coverage for all modified/added functions, components, and branches
      4. Include proper mocking of GraphQL queries/mutations, external dependencies, and API calls
      5. Test both user interactions and state changes comprehensively
      6. Require any additional tests for uncovered scenarios
   2. For each modified file with logic or UI, verify corresponding test files exist and are updated
   3. Check that test files follow naming conventions (*.test.tsx, *.test.ts, *.spec.tsx, *.spec.ts)
   4. Validate that tests don't include disabled tests (it.skip, describe.skip) unless explicitly documented
   5. Ensure mock data cleanup is performed (as required by pre-commit hooks)
2. **Code Quality & Repository Standards:**
   
   1. Verify TypeScript compliance:
      
      1. No use of `any` type without explicit justification
      2. Proper interface/type definitions for all props, state, and function parameters
      3. No TypeScript errors or warnings
   2. Validate GraphQL implementation:
      
      1. Queries/mutations are properly typed and follow existing patterns in src/GraphQl/
      2. No duplicate or redundant queries (check for pagination inconsistencies)
      3. Error handling is implemented for all GraphQL operations
   3. Check React component best practices:
      
      1. Functional components with proper hooks usage
      2. Material-UI component patterns are followed consistently
      3. No prop drilling; proper use of context when needed
      4. Proper cleanup in useEffect hooks
   4. Verify internationalization (i18n):
      
      1. All user-facing strings use i18next translation keys
      2. New translation keys are added to locale files
      3. No hardcoded English text in UI components
   5. Validate accessibility:
      
      1. Proper ARIA labels and roles
      2. Keyboard navigation support
      3. Screen reader compatibility
3. **Documentation & Code Comments:**
   
   1. Verify auto-generated documentation is updated (as enforced by Check-AutoDocs job)
   2. Check for adequate JSDoc comments on public functions and complex logic
   3. Ensure README.md or related docs are updated if new features are added
   4. Validate that Table of Contents is updated (as required by pre-commit hooks)
4. **Identify Issues:**
   
   1. Lines of code that are missing tests (list specific line numbers)
   2. Any unnecessary files that have been submitted (build artifacts, node_modules, .env files, IDE configs)
   3. Unused imports or dead code (as checked by knip)
   4. Console.log statements or debug code left in production files
   5. Files that should be excluded from the PR scope
5. **PR Requirements Validation:**
   
   1. Verify the PR meets the requirements of the linked issue:
      
      1. All acceptance criteria are addressed
      2. No out-of-scope changes are included
   2. Validate compliance with all previously suggested improvements and comments:
      
      1. All review comments have been addressed
      2. Requested changes have been implemented correctly
   3. Check PR metadata:
      
      1. Proper title and description
      2. Linked to appropriate issue(s)
      3. Target branch is correct (typically `develop`)
   4. Verify no sensitive information is exposed (as checked by Check-Sensitive-Files job)
6. **CI/CD Pipeline Compatibility:**
   
   1. Validate that code changes won't break any of the 17+ CI jobs
   2. Ensure pre-commit hook requirements are met:
      
      1. Code formatting (Prettier)
      2. Linting (ESLint)
      3. Type checking (TypeScript)
   3. Check that changes don't introduce performance regressions for the 12-shard test execution
7. **Security & Best Practices:**
   
   1. No security vulnerabilities introduced
   2. No exposed API keys, tokens, or credentials
   3. Proper input validation and sanitization
   4. No SQL injection or XSS vulnerabilities in any data handling
8. **Implementation Correctness:**
   
   * For the specific implementation file(s) being tested:
     
     * Verify the implementation logic is correct and free of bugs
     * Validate that queries target the correct database tables/entities
     * Check that foreign key relationships are correctly enforced
     * Confirm business logic constraints are properly implemented
     * Any bugs, logic errors, or incorrect implementations discovered in the file(s) under test MUST be fixed in this PR before approval
9. **Review Decision:**
   
   1. When critical issues are found, clearly state that changes are required in your response but DO NOT append  the APPROVED tag.
   2. Only append  the APPROVED tag when ALL criteria are satisfied and zero changes are required:
      
      1. Complete test coverage (≥95%)
      2. No code quality issues, security concerns, or bugs
      3. No bugs or logic errors in the implementation file(s) under test
      4. Full implementation of issue requirements
      5. Compliance with all previous feedback
      6. No unnecessary files
10. **Focus Areas:**

* Prioritize substantive issues over minor stylistic suggestions
* Focus on issues that would cause CI pipeline failures or production bugs
* Highlight any deviations from established codebase patterns

**Note:** Test-focused PRs should maintain clear scope boundaries. However, this does NOT exempt bugs in the specific file(s) being tested - those must be fixed in the same PR. Only bugs in unrelated files may be deferred to separate issues.

**Comment by rawadhossain** [comment]
@palisadoes Please take a look when you have a moment. Thanks.
